lib/travis/build/script/matlab.rb

```diff
@@ -0,0 +1,46 @@
+module Travis
+  module Build
+    class Script
+      class Matlab < Script
+        MATLAB_INSTALLER_LOCATION = 'https://ssd.mathworks.com/supportfiles/ci/ephemeral-matlab/v0/install.sh'.freeze
+        MATLAB_DEPS_LOCATION = 'https://ssd.mathworks.com/supportfiles/ci/matlab-deps/v0/install.sh'.freeze
+        MATLAB_START = 'matlab -batch'.freeze
+        MATLAB_COMMAND = "assertSuccess(runtests('IncludeSubfolders',true));".freeze
+
+        DEFAULTS = {
+          matlab: 'latest'
+        }.freeze
+
+        def export
+          super
+          sh.export 'TRAVIS_MATLAB_VERSION', release.shellescape,
+                    echo: false
+        end
+
+        def setup
+          super
+          # Execute helper script to install runtime dependencies
+          sh.raw "wget -qO- --retry-connrefused #{MATLAB_DEPS_LOCATION}" \
+                 ' | sudo -E bash -s -- $TRAVIS_MATLAB_VERSION'
+
+          # Invoke the ephemeral MATLAB installer that will make a MATLAB available
+          # on the system PATH
+          sh.raw "wget -qO- --retry-connrefused #{MATLAB_INSTALLER_LOCATION}" \
+                 ' | sudo -E bash -s -- --release $TRAVIS_MATLAB_VERSION'
+        end
+
+        def script
+          super
+          # By default, invoke the default MATLAB 'runtests' command
+          sh.cmd "#{MATLAB_START} \"#{MATLAB_COMMAND}\""
+        end
+
+        private
+  
+        def release
+          Array(config[:matlab]).first.to_s
```

Comment:
Do we want to validate the value given? Being specific will give you better control over what is valid, but could also mean constant maintenance.

Reply:
Yes! We're passing the release to the scripts that we're using to carry out the task of setting MATLAB up on the instance. Each script validates the provided release input and errors out if it's not valid 😄